src/Statement/StatementList.php

```diff
@@ -87,7 +87,13 @@ public function getPropertyIds() {
 	}
 
 	public function addStatement( Statement $statement ) {
-		$this->statements[] = $statement;
+		if ( $statement->getGuid() !== null
+			&& !array_key_exists( $statement->getGuid(), $this->statements )
+		) {
+			$this->statements[$statement->getGuid()] = $statement;
+		} else {
+			$this->statements[] = $statement;
```

Comment:
That would be an other class. The main purpose of **this** class is an ordered list of Statement objects, no matter what their GUID is.

That's why this is an RFC.